src/goto-symex/symex_target.h

```diff
@@ -87,78 +118,132 @@ class symex_targett
     const sourcet &source)=0;
 
-  // record output
+  /// Record an output.
+  /// \param guard: Precondition for writing to the output
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  output
+  /// \param output_id: Name of the output
+  /// \param args: A list of IO arguments
   virtual void output(
     const exprt &guard,
     const sourcet &source,
     const irep_idt &output_id,
     const std::list<exprt> &args)=0;
 
-  // record formatted output
+  /// Record formatted output.
+  /// \param guard: Precondition for writing to the output
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  output
+  /// \param output_id: Name of the output
+  /// \param fmt: Formatting string
+  /// \param args: A list of IO arguments
   virtual void output_fmt(
     const exprt &guard,
     const sourcet &source,
     const irep_idt &output_id,
     const irep_idt &fmt,
     const std::list<exprt> &args)=0;
 
-  // record input
+  /// Record an input.
+  /// \param guard: Precondition for reading from the input
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  input
+  /// \param input_id: Name of the input
+  /// \param args: A list of IO arguments
   virtual void input(
     const exprt &guard,
     const sourcet &source,
     const irep_idt &input_id,
     const std::list<exprt> &args)=0;
 
-  // record an assumption
+  /// Record an assumption.
+  /// \param guard: Precondition for reaching this assumption
+  /// \param cond: Condition this assumption represents
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  assumption
   virtual void assumption(
     const exprt &guard,
     const exprt &cond,
     const sourcet &source)=0;
 
-  // record an assertion
+  /// Record an assertion.
+  /// \param guard: Precondition for reaching this assertion
+  /// \param cond: Condition this assertion represents
+  /// \param msg: The message associated with this assertion
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  assertion
   virtual void assertion(
     const exprt &guard,
     const exprt &cond,
     const std::string &msg,
     const sourcet &source)=0;
 
-  // record a goto
+  /// Record a goto instruction.
+  /// \param guard: Precondition for reaching this goto instruction
+  /// \param cond: Condition under which this goto should be taken
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  goto instruction
   virtual void goto_instruction(
     const exprt &guard,
     const exprt &cond,
     const sourcet &source)=0;
 
-  // record a constraint
+  /// Record a _global_ constraint: there is no guard limiting its scope.
+  /// \param cond: Condition represented by this constraint
+  /// \param msg: The message associated with this constraint
+  /// \param source: Pointer to location in the input GOTO program of this
+  ///  constraint
```

Comment:
:question: I asked this before but it may have gotten lost - why are the last four functions below not documented?

Reply:
Added.